gobblin-hive-registration/src/main/java/gobblin/hive/HiveSerDeManager.java

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.hive;
+
+import java.io.IOException;
+
+import org.apache.commons.lang3.reflect.ConstructorUtils;
+import org.apache.hadoop.fs.Path;
+
+import com.google.common.base.Enums;
+import com.google.common.base.Optional;
+import com.google.common.base.Throwables;
+
+import gobblin.annotation.Alpha;
+import gobblin.configuration.State;
+import gobblin.hive.avro.HiveAvroSerDeManager;
+import lombok.extern.slf4j.Slf4j;
+
+
+/**
+ * This class manages SerDe properties (including schema properties) for Hive registration.
+ *
+ * @author ziliu
+ */
+@Slf4j
+@Alpha
+public abstract class HiveSerDeManager {
+
+  public static final String HIVE_ROW_FORMAT = "hive.row.format";
+
+  protected final State props;
+
+  protected HiveSerDeManager(State props) {
+    this.props = props;
+  }
+
+  /**
+   * Add the appropriate SerDe properties (including schema properties) to the given {@link HiveRegistrationUnit}.
+   */
+  public abstract void addSerDeProperties(Path path, HiveRegistrationUnit hiveUnit) throws IOException;
+
+  /**
+   * Update the schema in the existing {@link HiveRegistrationUnit} into the schema in the new
+   * {@link HiveRegistrationUnit}.
+   */
+  public abstract void updateSchema(HiveRegistrationUnit existingUnit, HiveRegistrationUnit newUnit) throws IOException;
```

Comment:
Not necessarily agree, since serde and schema are tightly related (if you use Avro serde then you need to use Avro schema), so it makes sense to have a class that manages both.